charts/redpanda/rbac.go

```diff
@@ -155,7 +155,9 @@ func ClusterRoleBindings(dot *helmette.Dot) []*rbacv1.ClusterRoleBinding {
 func SidecarControllersClusterRole(dot *helmette.Dot) *rbacv1.ClusterRole {
 	values := helmette.Unwrap[Values](dot.Values)
 
-	if !values.Statefulset.SideCars.Controllers.Enabled || !values.Statefulset.SideCars.Controllers.CreateRBAC {
+	if (!values.Statefulset.SideCars.Controllers.Enabled || !values.Statefulset.SideCars.Controllers.CreateRBAC) &&
+		!values.Statefulset.SideCars.PVCUnbinder.Enabled &&
+		!values.Statefulset.SideCars.BrokerDecommissioner.Enabled {
```

Comment:
NIT: To not have 4 places with complex condition you could create function in SideCars struct. 
```suggestion
	if !values.Statefulset.SideCars.IsEnabled() {
```